### PR TITLE
HOSTEDCP-1044: Add nodepools telemetry metrics for HyperShift

### PR DESCRIPTION
**Pull request by muraee**
- hypershift:nodepools:size
- hypershift:nodepools:available_replicas

requires: https://github.com/openshift/hypershift/pull/3593

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by muraee** [comment]
cc @simonpasquier

**Review by simonpasquier — COMMENTED** [review]
From https://github.com/openshift/hypershift/pull/3593#discussion_r1504465322

> changed to aggregate per HostedCluster, the maximum cardinality would between 80-100

So at most a single management cluster running the hypershift operator could generate 2 x 100 = 200 series. It's still way above the "automatically approved" limit (https://rhobs-handbook.netlify.app/products/openshiftmonitoring/telemetry.md/#request-approval).

cc @jan--f @moadz

**Comment by simonpasquier** [comment]
Another question: is there a need to correlate these metrics with the telemetry metrics emitted from the guest cluster?

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -921,6 +921,16 @@ data:
     # platform:hypershift_nodepools:max is the total number of nodepools managed by the hypershift operator by cluster platform
     - '{__name__="platform:hypershift_nodepools:max"}'
     #
+    # owners: (@openshift/team-hypershift-maintainers)
+    #
+    # cluster_name:hypershift_nodepools_size:sum is the total number of desired nodepool replicas managed by the hypershift operator per HostedCluster
```
[the remark this answers is not in the document]
Comment (muraee) [reply Documentation/data-collection.md:926]: I don't think so

**Comment by muraee** [comment]
> So at most a single management cluster running the hypershift operator could generate 2 x 100 = 200 series

per HostedCluster, multiple HostedClusters can be created in the same management cluster.

> Another question: is there a need to correlate these metrics with the telemetry metrics emitted from the guest cluster?

No need cc@zanetworker

**Comment by simonpasquier** [comment]
cc @zanetworker, see https://github.com/openshift/cluster-monitoring-operator/pull/2265#issuecomment-2092613336 (the previous mention failed).

File: Documentation/data-collection.md (hunk 1 above)
Comment (simonpasquier) [inline Documentation/data-collection.md:926]: (nit)
```suggestion
    # cluster_name:hypershift_nodepools_size:sum is the total number of desired nodepool replicas managed by the hypershift operator per HostedCluster identified by `the cluster_name` and `exported_namespace` labels.
```

**Comment by zanetworker** [comment]
>Another question: is there a need to correlate these metrics with the telemetry metrics emitted from the guest cluster?

No need @simonpasquier

File: Documentation/data-collection.md (hunk 1 above)
Comment (muraee) [reply Documentation/data-collection.md]: updated

**Comment by muraee** [comment]
@simonpasquier are we ok to move forward and merge this?

**Comment by jan--f** [comment]
Not sure if this was discussed elsewhere already, but for a telemetry addition of this size we require management buy in. Please get @eparis approval for this.

**Comment by zanetworker** [comment]
@eparis any comments/objections on this. This is needed to be able track cluster sizes in HyperShift.

**Comment by zanetworker** [comment]
@eparis Ping :)

**Comment by eparis** [comment]
Daniel reached out to me directly. I miss 99.9999%of github notifications  :-(     I approve.

**Pull request merged**